src/compiler/moduleNameResolver.ts

```diff
@@ -1577,9 +1577,9 @@ namespace ts {
      * that the runtime has already been executed).
      * @internal
      */
-    function isPnpAvailable() {
+    function isPnpAvailable(path: string) {
         // @ts-ignore
-        return process.versions.pnp;
+        return process.versions.pnp && getPnpApi().findPackageLocator(`${path}/`) !== null ? true : false;

```

Comment:
Let me know if you'd like me to change these to:
`return !!(process.versions.pnp && getPnpApi().findPackageLocator(`${path}/`));`